sql/item_pfsfunc.h

```diff
@@ -0,0 +1,55 @@
+#ifndef ITEM_PFS_FUNC_INCLUDED
+#define ITEM_PFS_FUNC_INCLUDED
+
+/* Copyright (c) 2000, 2022, Oracle and/or its affiliates.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License, version 2.0,
+   as published by the Free Software Foundation.
+
+   This program is also distributed with certain software (including
+   but not limited to OpenSSL) that is licensed under separate terms,
+   as designated in a particular file or component or in included license
+   documentation.  The authors of MySQL hereby grant you an additional
+   permission to link the program and your derivative works with the
+   separately licensed software that they have included with MySQL.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License, version 2.0, for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
+
+#include "item_func.h"      // Item_str_func, Item_int_func
+
+#ifdef USE_PRAGMA_INTERFACE
+#pragma interface			/* gcc class implementation */
+#endif
+
+/** format_pico_time() */
+
+class Item_func_pfs_format_pico_time : public Item_str_func {
+    typedef Item_str_func super;
+    String m_value;
+    char m_value_buffer[20];
```

Comment:
Can you set this to 12, since is said that `/* Format is 'AAAA.BB UUU' = 11 characters or 'AAA ps' = 6 characters. */`

Reply:
When trying to make, I gets error `‘__builtin___sprintf_chk’ output between 7 and 15 bytes into a destination of size 12`.
I made it of size 15

Reply:
Which compiler you are using? Can you trace this and find out exactly where that happens and update comment `11 characters ` that is wrong than.

Reply:
I am using gcc 11.3
It got resolved with `snprintf`, since we specified maximum to be 12.
I will re-look more into that to understand why, but I guess this could be because of maximum unit length & signs of the values/exponent. But according to the logic (with the given units), this should never happen and the length should be 12  char maximum (including the null terminator)